app/decorators/component.ts

```diff
@@ -0,0 +1,67 @@
+import Component from '@ember/component';
+import { assert } from '@ember/debug';
+
+/**
+ * Property decorator for required action arguments in components.
+ * Provides a default action method that raises a useful error.
+ *
+ * Usage:
+ *
+ * ```ts
+ *  import { requiredAction } from 'ember-osf-web/utils/required-action';
+ *
+ *  class MyComponent extends Component {
+ *      // Required arguments
+ *      @requiredAction doTheThing: (arg: number) => void;
+ *  }
+ * ```
+ */
+export function requiredAction(
+    classPrototype: any,
+    propertyName: string,
+): any {
+    const className = classPrototype.constructor.name;
+
+    // eslint-disable-next-line no-param-reassign
+    classPrototype[propertyName] = () => {
+        assert(`Missing required action in ${className}: ${propertyName}`);
+    };
+}
+
+/**
+ * Property decorator for optional action arguments in components.
+ * Provides a default action method that does nothing.
+ *
+ * Usage:
+ *
+ * ```ts
+ *  import { optionalAction } from 'ember-osf-web/utils/required-action';
+ *
+ *  class MyComponent extends Component {
+ *      // Optional arguments
+ *      @optionalAction doTheThing: (arg: number) => void;
+ *  }
+ * ```
+ */
+export function optionalAction(
+    classPrototype: any,
+    propertyName: string,
+): any {
+    // eslint-disable-next-line no-param-reassign
+    classPrototype[propertyName] = () => { /* empty */ };
+}
```

Comment:
I'm not sure how useful this is. It seems that, most of the time, when an action is optional, we will want to check if it is defined and this will prevent that check. I guess this would allow you to just always call the action, which would be sometimes useful and prevent the need for a defined check, but I think that could be confusing and, in some cases, prevent desired behavior, such as overriding a default action.

Reply:
Yeah, good point, I'm not so sure anymore this is useful. I like the symmetry with `@requiredAction` and clearly highlighting which component parameters were actions. But it doesn't make sense if we want default behavior, which is likely common... I guess I'll just delete this one.